File: C/Easy/108.c
// Daily Coding Problem: 108
// Difficulty: Easy
// Asked by: Google
// Authored by: TenthEdict

#include <stdio.h>
#include <string.h>
#include <stdbool.h>

bool checkStringRotation(const char* a, const char* b) {

    if (strlen(a) != strlen(b)) {
        return false;
    }

    char doubled[256];

    sprintf(doubled, "%s%s", a, a);

    if (strstr(doubled, b) == NULL) {
        return false;
    } else {
        return true;
    }
}

int main() {
    printf("%d\n", checkStringRotation("abcde", "cdeab"));  // 1 (true)
    printf("%d\n", checkStringRotation("abc", "acb"));      // 0 (false)
    printf("%d\n", checkStringRotation("", ""));            // 1 (true)
    printf("%d\n", checkStringRotation("abc", "abcd"));     // 0 (false)
    return 0;
}

File: Python/Easy/108.py
# Daily Coding Problem: 108
# Difficulty: Easy
# Asked by: Google
# Authored by: TenthEdict

def check_string_rotation(a, b):
    if len(a) != len(b):
        return False
    return b in a + a
    
print(check_string_rotation("abcde", "cdeab")) # True
print(check_string_rotation("abcde", "abcde")) # True
print(check_string_rotation("abc", "acb")) # False
print(check_string_rotation("abc", "abcd")) # False
print(check_string_rotation("","")) # True
print(check_string_rotation("a", "a")) # True
